[WIP] AbstractDiffractometer and AbstartSampleView

beteva opened the pull request:
Changes due to AbstractDiffractometer and AbstartSampleView new implementation. Dependant on
mxcube/mxcubecore/pull/1386

mockoocy committed:
Add CheckboxWidget

Closes #1896
The custom checkbox widget is used to provide the description tooltip
for the description fields.
By default RJSF renders the checkbox with the description and label
provided next to the checkbox, as text paragraph.
To make it more consistent, and reduce visual clutter, the description
will be rendered in a tooltip, as in other fields.

There is also a small CSS change, which makes the label title vertically
aligned with the description tooltip icon - it's also rendered directly
after the label.

Commit:
Changes due to AbstractDiffractometer and AbstartSampleView new implementation

walesch-yan committed:
[TMP] some changes to make it work

beteva committed:
Merge branch 'develop' into abstract_diffarctometer

jbflo committed:
Fix access component property

Commit:
Use sample_view instead of diffractometer

beteva committed:
Merge branch 'develop' into abstract_diffarctometer

axelboc committed:
Connect `SampleImage` to Redux store directly and bind sample view actions

axelboc committed:
Map state to props in `SampleImage` (except `sampleview` state)

axelboc committed:
Remove spreading of sample view state

axelboc committed:
Refactor derived state computations in `SampleViewContainer`

axelboc committed:
Return new shape from `addShape` thunk to avoid callback

Commit:
Cleaning again

marcus-oscarsson committed:
[skip ci] Bumped minor version

marcus-oscarsson committed:
[skip ci] Bumped minor version

marcus-oscarsson committed:
[skip ci] Bumped minor version

Phantominumm committed:
Remove Sonar Scan and upload bandit results in the Security tab

Change Bandit logic and Test with low parameters

return to Medium security level parameters

Update ruff.toml

Co-authored-by: fabcor <fabien.coronis@maxiv.lu.se>

Clean up bandit_sonar.yaml by removing comments

Removed commented-out SonarCloud scan steps from the workflow.

Change to bandit sarif and remove convertor

marcus-oscarsson committed:
[skip ci] Bumped minor version

axelboc committed:
Access Redux store from `PlateManipulator` to remove props

axelboc committed:
Access more specific slices of state

walesch-yan commented on mxcubeweb/core/adapter/diffractometer_adapter.py:
```diff
@@ -53,8 +55,8 @@ def state(self):
         return "READY" if self._ho.is_ready() else "BUSY"
 
     def head_configuration(self) -> dict:
-        data = self._ho.get_head_configuration()
-        return data.dict() if data else {}
+        data = self._ho.get_head_type.name  # noqa: F841
+        return {}
```
It seems that this functionality is now missing, so `get_head_configuration` does not exist on `AbstractDiffractometer`, but (only) needed for SSXChip

mockoocy committed:
Move element data to a json file

Currently the JSX for the `PeriodicTable` component is very long, as it
both defines the data and renders it.
It is ~800 lines of mostly repeated code.
I've added a json blob, with selected fields from
[there](https://github.com/Bowserinator/Periodic-Table-JSON/blob/master/PeriodicTableJSON.json).

There are also some very minor changes in variable names, etc.
and some major CSS changes, as the table is now rendered as a grid.

mockoocy committed:
Disable unavailable elements

Currently elements which are not set as available (via `energy_scan`
config file) have lower opacity.
They can still be selected (although selecting them will lead to an
invalid state of the form; the Element field will be filled, but Edge
field will not.
This change removes the possibility of selecting such fields.

marcus-oscarsson committed:
[skip ci] Bumped minor version

mockoocy committed:
Fix crash when field with ui schema has no group

Closes #1895
Currently code assumes that if field has an entry in uiSchema JSON, then
it has an entry for group - due to a missing nullish coalescence.

axelboc committed:
Fix regression in `SSXChipControl`

marcus-oscarsson committed:
[skip ci] Bumped minor version

marcus-oscarsson committed:
[skip ci] Bumped minor version

marcus-oscarsson committed:
[skip ci] Bumped minor version

marcus-oscarsson committed:
[skip ci] Bumped minor version

marcus-oscarsson committed:
[skip ci] Bumped minor version

axelboc committed:
Clean up `main.css`

marcus-oscarsson committed:
[skip ci] Bumped minor version

Commit:
change of definitions for geometries.
